students/alex_skrn/Lesson05Extra/logging_threading.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+"""Logging from multiple threads.
+
+To simplify things a bit I removed syslog logging and I still
+have one function only, so that not to confuse me even more.
+
+Am I moving in the right direction with what I have now?
+
+Issue 1: At this point I have troubles wrapping logging-related code into
+a function and running it repeatedly in the main block because
+in such case loggers multiply quickly and produce many repeated
+log records. This is why I keep all log definions global.
+
+Issue 2. Another unclear thing is that part about "Make sure the threads exit
+after its done with execution". Why would
+I need them after execution? I thought the purpose of threads was just
+to run some code within them. So that thread.start() launches a thread
+together with some function/program associated
+with it (i.e. threading(target=my_fun)), and as soon as the associated
+program is over, I don't need this thread any longer, isn't it?
+
+Issue 3. I don't understand what grep is. From what I see in the Internet,
+it is about using regular expressions for pattern matching, is that right?
+Do I need to use regexps at all to search for uuid? It seems that no two
+persons on the Internet can agree on the right regexp for uuid. Why
+can't I just put uuid identifier at the log beginning and then use it as a key
+like I do below?
```

Comment:
yep, thats exactly what i'm saying.  So your command would look something like `grep '0f49cac2-f424-484f-a747-aeeede0a3257' my_logs.log` on a linux computer.  `grep` should also be available on your mac.  You don't need to do any complicated regex though,  the above command should be all you need.